### PR TITLE
Update docs to be more strict around sign-in resolvers + remove insecure sign-in resolver

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Sign-in resolvers are very important from a security PoV and we've seen too much misconfiguration in the wild. This aims to highlight some of the common pitfalls and remove insecure examples/implementations to lower the risk of insecure configurations.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/overview/threat-model.md [hunk 1]
```diff
@@ -53,11 +53,11 @@ Note that the `UrlReaderService` system operates with a service context and is n
...
+As part of signing in with an sign-in resolver, a Backstage Token is issued containing the resolved user identity. The tokens are asymmetrically signed JSON Web Tokens, with the public keys available to any service that wishes to verify a token. The signing keys are rotated continuously and are unique to each installation of Backstage, meaning that Backstage Tokens are not shared across installations. The token contains claims for the user identity and ownership information, which can be used to determine what Backstage resources are owned by that user or group. It is important that this token can not be forged outside of the `auth` plugin, with the exception of other plugins deployed in the same backend service or sharing the same database. For a high-security deployment, the `auth` backend should therefore be deployed in a separate service with its own database.
```
Comment (freben) [inline docs/overview/threat-model.md:60]:
```suggestion
As part of signing in with a sign-in resolver, a Backstage Token is issued containing the resolved user identity. The tokens are asymmetrically signed JSON Web Tokens, with the public keys available to any service that wishes to verify a token. The signing keys are rotated continuously and are unique to each installation of Backstage, meaning that Backstage Tokens are not shared across installations. The token contains claims for the user identity and ownership information, which can be used to determine what Backstage resources are owned by that user or group. It is important that this token can not be forged outside of the `auth` plugin, with the exception of other plugins deployed in the same backend service or sharing the same database. For a high-security deployment, the `auth` backend should therefore be deployed in a separate service with its own database.
```

**Review by freben — APPROVED** [review]

**Comment by Rugvip** [comment]
Late addition: I added an `allowedOrigins` option for the local part resolver, and encourage used of that.

File: plugins/auth-backend-module-vmware-cloud-provider/report.api.md [hunk 2]
```diff
@@ -31,14 +29,6 @@ export interface VMwareCloudAuthenticatorContext {
   providerStrategy: Strategy;
 }
 
-// @public
-export namespace vmwareCloudSignInResolvers {
```
Comment (freben) [inline plugins/auth-backend-module-vmware-cloud-provider/report.api.md:35]: technically breaking - make minor?

File: plugins/auth-backend-module-oidc-provider/report.api.md [hunk 3]
```diff
@@ -39,7 +39,10 @@ export type OidcAuthResult = {
 export namespace oidcSignInResolvers {
   const emailLocalPartMatchingUserEntityName: SignInResolverFactory<
     unknown,
-    unknown
+    | {
+        allowedDomains?: string[] | undefined;
+      }
+    | undefined
```
Comment (freben) [inline plugins/auth-backend-module-oidc-provider/report.api.md:45]: seems possible to remove as well, similar to what you did for vmware cloud?

**Review by freben — APPROVED** [review]

File: plugins/auth-backend-module-vmware-cloud-provider/report.api.md (hunk 2 above)
Comment (Rugvip) [reply plugins/auth-backend-module-vmware-cloud-provider/report.api.md:35]: I did 😁  https://github.com/backstage/backstage/pull/26969/files#diff-d1c3662c7fcac2295578c0b0c488b2c3f2da65af8ddf5623d5b94cacb818505b

File: plugins/auth-backend-module-oidc-provider/report.api.md (hunk 3 above)
Comment (Rugvip) [reply plugins/auth-backend-module-oidc-provider/report.api.md:45]: Hmm, how do you mean? Get rid of the common email local part one altogether?

**Pull request merged**